### PR TITLE
Add documentation for read/write splitting Spring limitations

### PR DESCRIPTION
**Pull request by congoamz**
### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-## Read-Write Splitting Plugin
+# Read-Write Splitting Plugin
 
 The read-write splitting plugin adds functionality to switch between writer/reader instances via calls to the `Connection#setReadOnly` method. Upon calling `setReadOnly(true)`, the plugin will establish a connection to a reader instance and direct subsequent queries to this instance. Future calls to `setReadOnly` will switch between the established writer and reader connections according to the boolean argument you supply to the `setReadOnly` method.
```
Comment (brunos-bq) [inline docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md:3]: NIT: is it worth adding "a new connection to a randomly chosen reader instance"

File: docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md [hunk 2]
```diff
@@ -113,3 +113,14 @@ There are many session state attributes that can change during a session, and ma
 All other session state attributes will be lost when switching connections between the writer/reader.
 
 If your SQL workflow depends on session state attributes that are not mentioned above, you will need to re-configure those attributes each time that you switch between the writer/reader.
+
+
+### Spring limitations
```
Comment (brunos-bq) [inline docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md:118]: Known limitations when using Spring Boot/Framework

File: docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md [hunk 3]
```diff
@@ -113,3 +113,14 @@ There are many session state attributes that can change during a session, and ma
...
+
+#### @Transactional(readOnly = True)
+
+We recommend that you do not use the read/write splitting plugin in conjunction with Spring's @Transactional(readOnly = True) annotation. When a method with this annotation is hit, Spring calls conn.setReadOnly(true), executes the method, and then calls setReadOnly(false) to restore the connection's initial readOnly value. Consequently, every time the method is called, the plugin switches to the reader, executes the method, and then switches back to the writer. Although the reader connection will be cached after the first setReadOnly call, there is still some overhead when switching between the cached writer/reader connections. This constant switching is not an ideal use of the plugin because it is frequently incurring this overhead. The suggested approach for this scenario is to avoid loading the read/write splitting plugin and instead use the writer cluster URL for your write operations and the reader cluster URL for your read operations. By doing this you avoid the overhead of constantly switching between connections while still spreading load across the database instances in your cluster.
```
Comment (brunos-bq) [inline docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md:122]: WARNING: The use of RWSplitting with the annotation @Transactional(readOnly = True) is not recommended.

When...

File: docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md (hunk 1 above)
Comment (congoamz) [reply docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md:3]: which reader is chosen depends on the reader selection strategy (could be random, could be the reader with the least connections), but I'll update to clarify this

File: docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md [hunk 4]
```diff
@@ -69,47 +69,60 @@ private static String getPoolKey(HostSpec hostSpec, Properties props) {
...
 All other session state attributes will be lost when switching connections between the writer/reader.
 
 If your SQL workflow depends on session state attributes that are not mentioned above, you will need to re-configure those attributes each time that you switch between the writer/reader.
+
+
+### Limitations when using Spring Boot/Framework
+
+#### @Transactional(readOnly = True)
+
+> :warning: The use of read/write splitting with the annotation @Transactional(readOnly = True) is not recommended.
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md:122]: > [!WARNING] 
>

**Review by sergiyvamz — APPROVED** [review]

**Review by karenc-bq — APPROVED** [review]

**Pull request merged**